workflow-runtime/src/main/java/com/squareup/workflow/internal/NoOutput.kt

```diff
@@ -0,0 +1,21 @@
+package com.squareup.workflow.internal
+
+/**
+ * Sentinel value that, when used as an `OutputT` value, indicates that a [WorkflowAction] did not
+ * emit any output.
+ */
+internal object NoOutput {
+  /**
+   * Runs [block] with this value and returns the block's value, unless this value is [NoOutput],
+   * in which case it returns [NoOutput]. This is similar to an `Optional.map` operation.
+   */
+  @Suppress("UNCHECKED_CAST")
+  inline fun <T, R> T.letUnlessNoOutput(block: (T) -> R): R =
+    if (this === NoOutput) NoOutput as R else block(this)
+}
+
+/**
+ * Helper function to return [NoOutput] cast to [T].
+ */
+@Suppress("UNCHECKED_CAST", "NOTHING_TO_INLINE")
+internal inline fun <T> noOutput(): T = NoOutput as T
```

Comment:
How on earth does this work?

Reply:
Generics are erased, you can cast any type to a generic.